### PR TITLE
Show error text when a device name choice is invalid

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

The error message I used ("Your device name must have 3-64 characters and not end with punctuation.") is not precise, since there are some punctuation characters that can't be used at all, but the actual rules seem too complex to try to explain in a sentence.

Screen Shot 2020-03-17 at 11 43 23 AM

### CODE REVIEW

File: shared/provision/set-public-name/index.tsx [hunk 1]
```diff
@@ -19,6 +19,15 @@ type Props = {
 const SetPublicName = (props: Props) => {
   const [deviceName, setDeviceName] = React.useState(defaultDevicename)
   const cleanDeviceName = Constants.cleanDeviceName(deviceName)
+  const disabled = React.useMemo(() => {
```
Comment (chrisnojima) [inline shared/provision/set-public-name/index.tsx]: is there an advantage to memoing this as this'll get called as you're typing and you'll only skip if somehow this rerenders w/o it being due to the devicename? also likely not expensive to just calc it

File: shared/provision/set-public-name/index.tsx [hunk 2]
```diff
@@ -62,13 +71,21 @@ const SetPublicName = (props: Props) => {
         <Kb.Box2 direction="vertical" style={styles.wrapper} gap="xsmall">
           <Kb.NewInput
```
Comment (jakob223) [inline shared/provision/set-public-name/index.tsx]: how about capping this input at 64

File: shared/provision/set-public-name/index.tsx [hunk 3]
```diff
@@ -62,13 +71,21 @@ const SetPublicName = (props: Props) => {
...
             autoFocus={true}
+            error={disabled}
             placeholder="Pick a device name"
             onEnterKeyDown={onSubmit}
             onChangeText={setDeviceName}
```
Comment (jakob223) [inline shared/provision/set-public-name/index.tsx]: we could just filter out illegal characters here, by passing it through a filter before calling `setDeviceName`

**Comment by cjb** [comment]
Thanks both, made these changes.

**Pull request merged**